File: src/leetcode/reverse-prefix-of-word/reverse-prefix-of-word.js
/**
 * Leetcode #2000
 * Difficulty: Easy
 * https://leetcode.com/problems/reverse-prefix-of-word/
 */
/**
 * @param {string} word
 * @param {string} ch
 * @return {string}
 */
export const reversePrefix = (word, ch) => {
  let reversed = '';
  let hasReversed = false;

  for (let i = 0; i < word.length; i += 1) {
    if (hasReversed) {
      reversed += word[i];
    } else {
      reversed = word[i] + reversed;
    }

    if (word[i] === ch) {
      hasReversed = true;
    }
  }

  return hasReversed ? reversed : word;
};

File: src/leetcode/reverse-prefix-of-word/reverse-prefix-of-word.rs
impl Solution {
    pub fn reverse_prefix(word: String, ch: char) -> String {
        let mut reversed = false;

        word.chars().fold(String::from(""), |mut acc, c| {
            acc += &*c.to_string();

            if !reversed && c == ch {
                reversed = true;
                acc = acc.chars().rev().collect::<String>();
            }

            acc
        })
    }
}